front/src/applications/operationalStudies/hooks/useInvalidTrainOps.ts

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from 'react';
+
+import type {
+  PathPropertiesFormatted,
+  SimulationResults,
+} from 'applications/operationalStudies/types';
+import {
+  osrdEditoastApi,
+  type OperationalPointReference,
+  type OperationalPointPart,
+} from 'common/api/osrdEditoastApi';
+
+import { getStationFromOps, isOperationalPointReference } from '../utils';
+
+const useInvalidTrainOps = (
+  infraId: number,
+  simulationResults: SimulationResults | undefined
+): PathPropertiesFormatted['operationalPoints'] => {
+  const operationalPointReferences: OperationalPointReference[] = useMemo(() => {
+    if (!simulationResults || simulationResults.isValid) return [];
+    return simulationResults.train.path
+      .filter((pathItem) => isOperationalPointReference(pathItem))
+      .map(
+        ({ id: _id, deleted: _deleted, ...cleanOperationalPointReference }) =>
+          cleanOperationalPointReference
+      );
+  }, [simulationResults]);
```

Comment:
We could avoid to iterate twice with `filter`an `map`like this
```suggestion
   return simulationResults.train.path.reduce((acc, pathItem) => {
  if (isOperationalPointReference(pathItem)) {
    const { id: _id, deleted: _deleted, ...cleanOperationalPointReference } = pathItem;
    acc.push(cleanOperationalPointReference);
  }
  return acc;
}, [] as OperationalPointReference[]);
```

Reply:
@emersion you didn't see this comment ❌

Reply:
> [!TIP]
>   To avoid the cast in the `reduce`, you can also do : 
```suggestion
   return simulationResults.train.path.reduce<OperationalPointReference[]>((acc, pathItem) => {
  if (isOperationalPointReference(pathItem)) {
    const { id: _id, deleted: _deleted, ...cleanOperationalPointReference } = pathItem;
    acc.push(cleanOperationalPointReference);
  }
  return acc;
}, []);
```

Reply:
Done!